src/main/java/uk/gov/cshr/civilservant/controller/OrganisationalUnitController.java

```diff
@@ -116,6 +122,21 @@ public ResponseEntity saveAgencyToken(@PathVariable Long organisationalUnitId, @
         }).orElseGet(() -> new ResponseEntity<>(HttpStatus.BAD_REQUEST));
     }
 
+    @PostMapping("/addOrganisationReportingPermission/{uid}")
+    @PreAuthorize("isAuthenticated()")
+    public ResponseEntity addOrganisationReportingPermission(@PathVariable String uid, @Valid @RequestBody ArrayList<String> organisationIds) {
+        Optional<CivilServant> civilServant = civilServantRepository.findByIdentity(uid);
+        //The below may not be required or if civilservant not present then return null.
+        if (!civilServant.isPresent()) {
+            return ResponseEntity.ok("Civil servant for this UID not found in database");
```

Comment:
This is a error scenario, hence it should return 500 error code rather then 200 OK.

Reply:
500 should only be returned if there was an actual issue on the server which meant it could not fulfil the request.  So check what server code is sent back and display accordingly ie if 200 and null then the response was OK but civil servant no in DB, if 500 returned then there was a problem in sending the response and display the appropriate msg

Reply:
It seems here that the system is not able to fetch user details for a given uid i.e. a loggedin user so it suggests that system has failed which is a 500 scenario.

Reply:
We should never assume the returned code, we should capture it and display appropriate message for that code in case statement.

Reply:
I agree Vince.

Reply:
I think 500 is server error and this error is more related to record not found. 500 is more when the server could not process the request.
server response is ok, its just that the records could not be found. I agree with error message to display however all error message across this functionality is under decision progress and should be decided by business ( and hence Mo for us).